src/networking.c

```diff
@@ -4725,6 +4726,9 @@ int processIOThreadsReadDone(void) {
         connSetPostponeUpdateState(c->conn, 0);
         connUpdateState(c->conn);
 
+        /* No client's data was read only TLS handshake. */
+        if (c->read_flags & READ_FLAGS_TLS_NEGOTIATION) continue;
+
```

Comment:
I do not like the fact that we mix io-threads and openssl specific logic with generic io-threads logic. maybe we can place a dedicated function which will default to always return C_OK when USE_OPENSSL is not defined?

Reply:
Changed to check for the conn state.